src/schema/v2/me/mark_all_notifications_as_read_mutation.ts

```diff
@@ -0,0 +1,36 @@
+import { GraphQLBoolean } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+
+export const markAllNotificationsAsReadMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "MarkAllNotificationsAsRead",
+  description: "Mark all unread notifications as read",
+  inputFields: {},
+  outputFields: {
+    success: {
+      type: GraphQLBoolean,
+      resolve: (result) => result.success,
+    },
+  },
+  mutateAndGetPayload: async (_, { updateNotificationsLoader }) => {
+    if (!updateNotificationsLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      await updateNotificationsLoader({ status: "read" })
+
+      return {
+        success: true,
+      }
+    } catch {
```

Comment:
**question**: should we call `Sentry.captureException` here? otherwise the error will not be logged anywhere and only [warning will be outputted](https://github.com/artsy/metaphysics/blob/86b39d687fb655079bd287fb83b31f67aa07812f/src/lib/loaders/api/loader_with_authentication_factory.ts#L68).

Reply:
or is it even better to use a common response format for mutations (for example, as in the screenshot), as is done for all mutations in Metaphysics?

![image](https://user-images.githubusercontent.com/3513494/190124214-93a8df5b-70ea-4cc9-afae-d25827d03972.png)

Reply:
I don't think we currently even use Sentry for Metaphysics as it is super noisy, and duplicates errors logged elsewhere in the request. The MP project in Sentry is even empty - https://sentry.io/organizations/artsynet/issues/?project=299996

In terms of mutation success/error patterns, yea I think the `... on MutationSuccess` and `... on MutationFailure` style is the way to go.

Reply:
> In terms of mutation success/error patterns, yea I think the ... on MutationSuccess and ... on MutationFailure style is the way to go

@dblandin what do you think about it? or is it better to keep the current solution with `success: true | false`?

Reply:
Sticking with the established pattern sounds good to me! We don't have much in terms of success attributes or anticipated errors to return, but I suppose this lets us easily add those later on within the response union type.

Reply:
Done